clients/src/main/java/org/apache/kafka/clients/consumer/internals/PrototypeAsyncConsumer.java

```diff
@@ -1093,6 +1099,10 @@ private void subscribeInternal(Collection<String> topics, Optional<ConsumerRebal
             log.info("Subscribed to topic(s): {}", join(topics, ", "));
             if (subscriptions.subscribe(new HashSet<>(topics), listener))
                 metadata.requestUpdateForNewTopics();
+
+            // Trigger subscribe event to effectively join the group if not already part of it,
+            // or just send the new subscription to the broker.
+            applicationEventHandler.add(new SubscriptionChangeApplicationEvent());
```

Comment:
I see, I was just intentionally leaving out all the pattern based logic because we don't support it at this point. But this makes me realize that that `subscribeInternal` based on pattern that you mentioned is wired to the `subscribe(Pattern pattern)` API call, when it's truly not supported yet. I think we should disable all the subscribe based on patterns until we implement them properly. What do you think?